### PR TITLE
Feature/minimal account center

### PR DESCRIPTION
**Pull request by mahmud-bn**
### Checklist
- [x] The version field in `package.json` is incremented following [semantic versioning](https://semver.org/)
- [x] The box that allows repo maintainers to update this PR is checked
- [x] I tested locally to make sure this feature/fix works
- [ ] This PR passes the Circle CI checks

### CODE REVIEW

File: packages/core/src/views/account-center/Micro.svelte
Comment (Adamj1232) [inline packages/core/src/views/account-center/Micro.svelte:52]: I think we have a border radius variable we could use here

Comment (mahmud-bn) [reply packages/core/src/views/account-center/Micro.svelte]: Thanks for the suggestion. Added a new one for 16px.

**Comment by gesquinca** [comment]
How would a dev configure which component they would like to use for the minimized state of account center? 
And how would they configure different ones for desktop and mobile?

**Comment by gesquinca** [comment]
is there a way to test this on the demo?

File: packages/core/src/views/account-center/Maximized.svelte
Comment (aaronbarnardsound) [inline packages/core/src/views/account-center/Maximized.svelte:48]:
```suggestion
  let innerWidth = 0
```
This can be a regular let binding rather than a reactive one.

File: packages/core/src/views/account-center/Maximized.svelte
Comment (aaronbarnardsound) [inline packages/core/src/views/account-center/Maximized.svelte:211]: Curious as to why the animation was removed here?

File: packages/core/src/views/account-center/Maximized.svelte
Comment (aaronbarnardsound) [inline packages/core/src/views/account-center/Maximized.svelte:229]: Would it be more robust to use our device check function with `bowser` to detect if on mobile?

File: packages/core/src/views/account-center/Micro.svelte
Comment (aaronbarnardsound) [inline packages/core/src/views/account-center/Micro.svelte:1]: Is there a reason for renaming this file to `Micro.svelte`? It ruins the diff for what has changed on this file, making it harder to review.

File: packages/core/yarn.lock
Comment (aaronbarnardsound) [inline packages/core/yarn.lock:1]: There shouldn't be a `yarn.lock` file in the core folder as the monorepo uses the one at the root of the project, so this will need to be removed.

File: packages/demo/.gitignore
Comment (aaronbarnardsound) [inline packages/demo/.gitignore:4]: We need the yarn.lock file to be checked in to the repo, so please remove this from the gitignore.

**Review by aaronbarnardsound — CHANGES_REQUESTED** [review]
Added some requested changes from looking at the code. 

Also I can't see in the diff changes to the Onboard initialization code that allow for `accountCenter` mobile configuration and the extra field in the desktop configuration and the associated validation?

**Comment by mahmud-bn** [comment]
> Added some requested changes from looking at the code.
> 
> Also I can't see in the diff changes to the Onboard initialization code that allow for `accountCenter` mobile configuration and the extra field in the desktop configuration and the associated validation?

File: packages/core/src/views/account-center/Micro.svelte
Comment (mahmud-bn) [reply packages/core/src/views/account-center/Micro.svelte:1]: `Micro.svelte` is a copy of `Minimized.svelte` — the same file with a bit fewer style customizations. I left the initial `Minimized.svelte` because I wasn't sure if we would need it for future configurations.

File: packages/core/src/views/account-center/Maximized.svelte
Comment (mahmud-bn) [reply packages/core/src/views/account-center/Maximized.svelte:211]: Removing the animations was necessary for achieving the smooth transition required in the Figma design. Tweaking the animation didn't achieve the required effect. The configurations can always switch to the `Minimized.svelte` file if the micro view is not wanted.

File: packages/core/src/views/account-center/Index.svelte
Comment (Adamj1232) [inline packages/core/src/views/account-center/Index.svelte:7]: Mismatch on naming here

File: packages/core/src/views/account-center/Maximized.svelte
Comment (mahmud-bn) [reply packages/core/src/views/account-center/Maximized.svelte:229]: Replaced it with the bowser mobile detection.

File: packages/core/src/views/account-center/Maximized.svelte
Comment (mahmud-bn) [reply packages/core/src/views/account-center/Maximized.svelte:211]: I will most likely return the animation back since the new minimal center will default to` false` for desktop.

**Comment by gesquinca** [comment]
The width of the open account center component is different depending on which minimized component you use. This should not be the case.

File: packages/core/src/types.ts
Comment (Adamj1232) [inline packages/core/src/types.ts:127]: Remove commented out code when you have a min

File: packages/core/src/validation.ts
Comment (Adamj1232) [inline packages/core/src/validation.ts:117]: remove empty line & at ln123

File: packages/core/src/types.ts
Comment (mahmud-bn) [reply packages/core/src/types.ts:127]: Done

File: packages/react/package.json
Comment (mahmud-bn) [reply packages/react/package.json:23]: Done

File: packages/core/src/validation.ts
Comment (mahmud-bn) [reply packages/core/src/validation.ts:117]: Done. Thanks

File: packages/core/src/views/account-center/Micro.svelte
Comment (Adamj1232) [inline packages/core/src/views/account-center/Micro.svelte:6]: There seems to be a handful of unused import, ` fade, shortenAddress, shortenEns, unrecognizedChainStyle`

Comment (Adamj1232) [inline packages/core/src/views/account-center/Micro.svelte:6]: There seems to be a handful of unused import, ` fade, shortenAddress, shortenEns, unrecognizedChainStyle`

File: packages/core/src/views/account-center/Micro.svelte
Comment (Adamj1232) [inline packages/core/src/views/account-center/Micro.svelte:23]: Doesn't look like we are using this anymore

Comment (Adamj1232) [inline packages/core/src/views/account-center/Micro.svelte:23]: Doesn't look like we are using this anymore

File: packages/core/src/views/account-center/Micro.svelte
Comment (Adamj1232) [inline packages/core/src/views/account-center/Micro.svelte:30]: Doesn't look like we are using this anymore

Comment (Adamj1232) [inline packages/core/src/views/account-center/Micro.svelte:30]: Doesn't look like we are using this anymore

File: packages/core/src/views/account-center/Micro.svelte
Comment (Adamj1232) [inline packages/core/src/views/account-center/Micro.svelte:36]: Are we still using this?

Comment (Adamj1232) [inline packages/core/src/views/account-center/Micro.svelte:36]: Are we still using this?

File: packages/core/README.md
Comment (Adamj1232) [inline packages/core/README.md:177]: Merge conflict to be resolved

**Review by Adamj1232 — APPROVED** [review]
Looks good! Good work sticking it out!
I would wait for @aaronbarnardsound for a final approval but code looks clean and works well from what I can tell

**Comment by gesquinca** [comment]
I'm seeing the `Micro.svelte` component rendering with a size of 84 x 56 px. This designs specify 80 x 56 px. 

I'm also still seeing `Mimimized.svelte` and `Maximized.svelte` components rendering with different widths. This should not be the case. They should have the same width.

File: packages/core/src/views/account-center/Index.svelte
Comment (aaronbarnardsound) [inline packages/core/src/views/account-center/Index.svelte:32]: I think that we want to set `width: 348px` here so that the account center widget is the correct width. The designs are for `316px` width, but we add `32px` for padding on either side.

Comment (aaronbarnardsound) [reply packages/core/src/views/account-center/Index.svelte:32]: After chatting to @gesquinca we want this container width to be `100%` and then we need a dynamic style on the container based on the device type:
```javascript
<div
  class="container flex flex-column absolute"
  style={`${accountCenterPositions[settings.position]} ${device.type === 'mobile' ? 'max-width: 352px;' : ''}`}
>
  {#if !settings.expanded && !settings.minimal}
    <!-- minimized -->
    <Minimized />
  {:else if !settings.expanded && settings.minimal}
    <!-- micro -->
    <Micro />
  {:else}
    <!-- maximized -->
    <Maximized />
  {/if}
</div>
```
The goal is that on mobile devices that the UI will cover the full width of the device (minus padding on either side) and then on anything other than mobile devices it will have a width of 352px (320px width plus 16px of padding on either side).

Comment (aaronbarnardsound) [reply packages/core/src/views/account-center/Index.svelte:32]: Actually, going for a more dynamic approach we can use media queries instead:

```css
.container {
    padding: 16px;
    font-family: var(--onboard-font-family-normal, var(--font-family-normal));
    width: 100%;
  }
  
  @media all and (min-width: 428px) {
    .container {
      max-width: 352px;
    }
  }
```

File: packages/core/src/constants.ts
Comment (mahmud-bn) [reply packages/core/src/constants.ts:12]: No, it's not. 
```
accountCenter: {
    enabled: true,
    position: 'topRight',
    expanded: false,
    minimal: getDevice().type === 'mobile'
  }
```

Comment (Adamj1232) [reply packages/core/src/constants.ts:12]: I think he is talking about the `locale` prop

Comment (mahmud-bn) [reply packages/core/src/constants.ts:12]: Ha! I see.

**Review by aaronbarnardsound — APPROVED** [review]
Looks good to me! 🎉

**Comment by funiimunii** [comment]
The current validation doesnt allow for setting minimal


```
const initOptions = Joi.object({
    wallets: walletInit,
    chains: chains.required(),
    appMetadata: appMetadata,
    i18n: Joi.object().unknown(),
    accountCenter: Joi.object({
        desktop: Joi.object({
            enabled: Joi.boolean(),
            position: accountCenterPosition
        }),
        mobile: Joi.object({
            enabled: Joi.boolean(),
            position: accountCenterPosition
        })
    })
});
```

**Comment by mahmud-bn** [comment]
> The current validation doesn't allow for setting minimal

Hi @funiimunii, 
We do have the current validation for `minimal`. You can look at the changes tab. We've also had it merged in the [main branch:  ](https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard/packages/core/src/validation.ts)

```
const initOptions = Joi.object({
  wallets: walletInit,
  chains: chains.required(),
  appMetadata: appMetadata,
  i18n: Joi.object().unknown(),
  accountCenter: Joi.object({
    desktop: Joi.object({
      enabled: Joi.boolean(),
      minimal: Joi.boolean(),
      position: accountCenterPosition
    }),
    mobile: Joi.object({
      enabled: Joi.boolean(),
      minimal: Joi.boolean(),
      position: accountCenterPosition,
    })
  })
}) 

```

**Comment by funiimunii** [comment]
aghh okay. we are using @web3-onboard/react npm package 2.1.6. Will upgrade to 2.1.7 and hopefully these made the cut?

**Pull request merged**
